x-pack/plugins/maps/public/classes/layers/new_vector_layer_wizard/wizard.tsx

```diff
@@ -60,18 +62,40 @@ export class NewVectorLayerEditor extends Component<RenderWizardArguments, State
     }
   }
 
-  _setCreateIndexError(errorMessage: string) {
+  _setCreateIndexError(errorMessage: string, userHasIndexWritePermissions: boolean = true) {
     if (!this._isMounted) {
       return;
     }
     this.setState({
       createIndexError: errorMessage,
+      userHasIndexWritePermissions,
+    });
+  }
+
+  async _checkIndexPermissions() {
+    return await getFileUpload().hasImportPermission({
+      checkCreateIndexPattern: true,
+      checkHasManagePipeline: false,
+      indexName: this.state.indexName,
     });
   }
 
   _createNewIndex = async () => {
     let indexPatternId: string | undefined;
     try {
+      const userHasIndexWritePermissions = await this._checkIndexPermissions();
+      if (!userHasIndexWritePermissions) {
+        this._setCreateIndexError(
+          i18n.translate('xpack.maps.layers.newVectorLayerWizard.indexPermissionsError', {
+            defaultMessage: 'User missing permissions to create index "{indexName}"',
```

Comment:
How about a message that is more similar to[ file uploads permission failure message](https://github.com/elastic/kibana/blob/master/x-pack/plugins/file_upload/public/components/import_complete_view.tsx#L108) `You do not have permission to create index "{indexName}".`